File: dev/magnets/magnets.py
#!/usr/bin/env python3

import sys
lines = sys.stdin.read().splitlines()
n = int(lines[0])
magnets = list(map(int, lines[1:]))

prev = 0
count = 0
for m in range(n):
    if prev != magnets[m]
        count += 1
    prev = magnets[m]

print(count)

File: dev/magnets/magnets.cpp
#include <bits/stdc++.h>

auto main() -> int {
  int n; std::vector<int> v;
  std::cin >> n;
  int x;
  while (std::cin >> x) {
    v.push_back(x);
  }

  int prev = 0; int count = 0;
  for (auto& m : v) {
    if (prev != m) {
      count++;
    }
    prev = m;
  }
  std::cout << count << "\n";
}
